driver-kotlin-sync/src/integration/kotlin/com/mongodb/kotlin/client/syncadapter/SyncListCollectionNamesIterable.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kotlin.client.syncadapter
+
+import com.mongodb.client.ListCollectionNamesIterable as JListCollectionsIterable
+import com.mongodb.kotlin.client.ListCollectionNamesIterable
+import java.util.concurrent.TimeUnit
+import org.bson.BsonValue
+import org.bson.conversions.Bson
+
+internal class SyncListCollectionNamesIterable(val wrapped: ListCollectionNamesIterable) :
```

Comment:
Is based on `com.mongodb.kotlin.client.syncadapter.SyncListCollectionsIterable`.